### ISSUE TITLE
[feature support] Enable to add affinities.

### ISSUE THREAD

**Issue by monaka** [teamhephy/workflow#147]
I want to set a Pod placement options in my multi-purpose K8s clusters. Especially, `affinity:`.

I created a patch against this issue. Could you accept and merge this ?

https://github.com/pizzafactory-bancallera/workflow-chart/commit/d2b9a43887976dfb4c638365f7578b0c1c90c4db

**Comment by Cryptophobia** [comment]
@monaka , I do not see issue with this. I don't think it will break anything. @kingdonb , thoughts?

If you make the PR, we can review and test it out. If it works for you with these changes it should work for all deployment type components.

**Comment by monaka** [comment]
@Cryptophobia Thanks for your review.
I wanna making PR based on this issue.
And, which repo should I send PRs for the workflow chart ?

**Comment by Cryptophobia** [comment]
@monaka , that is a good question. We need to make this as a PR to the workflow dir (but only update the values.yaml change and any of the umbrella chart helpers here). We also need to make a PR for each of the components `chart/` directories. Since this is the umbrella chart we just pull all the children charts into here and compile this chart.

This is quite a bit of tedious work as it requires a PR into all the repos that have a chart with such deployment. We also need to add the affinity section to the values of each of the components value.yaml of their charts as well.

This can be scripted out for speed. For example, here is a script that can do this at a high level if you want to follow this: https://github.com/teamhephy/docker-base/blob/master/_scripts/make-prs.sh

The idea will be to do the same and only edit the `*-deployment.yaml` and `values.yaml` for each of those files in the `chart/` directories.

**Comment by monaka** [comment]
@Cryptophobia I created PR on related projects.
And my question is not "how to pushing to multiple projects." but "Where is 'workflow' project that is maintained now?".
I guess code in this repository is not updated.

**Comment by Cryptophobia** [comment]
> @Cryptophobia I created PR on related projects.

Thank you @monaka !

> but "Where is 'workflow' project that is maintained now?". I guess code in this repository is not updated.

We have not made a new release in a while. I have some personal commitments that end this Monday (exam). And I hope to make a new release soon after that. I will notify you here when I merge all the PRs and update the charts for the new release. Likely I will stabalize one minor release and start working on new major version number.

**Comment by Cryptophobia** [comment]
You are right about most updated charts. The most updated chart for current release you can find at `charts.teamhephy.com` . There `workflow` or `workflow-beta` are the most updated charts. We only use the charts here as reference. When the main charts on the charts server above become out-of-sync, we sync them to what is in these directories here so the chart directories here should be source-of-truth.
